realm/kotlin-extensions/src/androidTest/kotlin/io/realm/KotlinRealmModelTests.kt

```diff
@@ -0,0 +1,233 @@
+
+    @get:Rule
+    val looperThread = RunInLooperThread()
+
+
+    private lateinit var realm: Realm
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getTargetContext())
+        realm = Realm.getInstance(configFactory.createConfiguration())
+    }
+
+    @After
+    fun tearDown() {
+        realm.close()
+    }
+
+    @Test
+    fun deleteFromRealm() {
+
+        // Make sure starting with 0
+        Assert.assertEquals(0, realm.where<SimpleClass>().count())
+
+        // Add 1, check count
+        realm.executeTransaction { it.createObject<SimpleClass>() }
+        Assert.assertEquals(1, realm.where<SimpleClass>().count())
+
+        // Delete the first, check count again.  !! is intentional to make
+        // sure we are sure calling deleteFromRealm
+        realm.executeTransaction { realm.where<SimpleClass>().findFirst()!!.deleteFromRealm() }
+        Assert.assertEquals(0, realm.where<SimpleClass>().count())
+
+    }
+
+    @Test
+    fun isValid() {
+        realm.executeTransaction {
+            val simple = it.createObject<SimpleClass>()
+            assertTrue("Expected valid after insert", simple.isValid())
+
+            simple.deleteFromRealm()
+            assertFalse("Expected invalid after delete", simple.isValid())
+        }
+    }
+
+   @Test
+    fun isManaged() {
+        realm.executeTransaction {
+            var simple = SimpleClass()
+            assertFalse("Expected not managed until attached", simple.isManaged())
+
+            simple = it.copyToRealm(simple)
+            assertTrue("Expected managed after attaching", simple.isManaged())
+        }
+    }
+
+    @Test
+    @RunTestInLooperThread
+    fun add_realmObjectChangeListener() {
+        val realm = looperThread.realm
+        realm.beginTransaction()
+        val simple = realm.createObject<SimpleClass>()
+        realm.commitTransaction()
+
+        looperThread.keepStrongReference(simple)
+        simple.addChangeListener(RealmObjectChangeListener{ simpleClass, changes ->
+            assert(changes?.isFieldChanged(SimpleClass::name.name) ?: false)
+            assertEquals("simple1", simpleClass.name)
+            looperThread.testComplete()
+        })
+
+        realm.beginTransaction()
+        simple.name = "simple1"
+        realm.commitTransaction()
+    }
+
+    @Test
+    @RunTestInLooperThread
+    fun add_realmChangeListener() {
+        val realm = looperThread.realm
+        realm.beginTransaction()
+        val simple = realm.createObject<SimpleClass>()
+        realm.commitTransaction()
+
+        looperThread.keepStrongReference(simple)
+        simple.addChangeListener(RealmChangeListener{ simpleClass ->
+            assertEquals("simple1", simpleClass.name)
+            looperThread.testComplete()
+        })
+
+        realm.beginTransaction()
+        simple.name = "simple1"
+        realm.commitTransaction()
+    }
+
+    @Test
+    @RunTestInLooperThread
+    fun remove_realmChangeListener() {
```

Comment:
This would just be `removeChangeListener`. Similar renames needed for the rest of these methods